test/test_parser.rb

```diff
@@ -2,20 +2,122 @@
 require 'rjson/parser'
 require 'rjson/tokenizer'
 require 'rjson/stream_tokenizer'
+require 'json'
 require 'stringio'
 
 module RJSON
   class TestParser < MiniTest::Unit::TestCase
-    def test_array
-      parser = new_parser '["foo",null,true]'
-      r = parser.parse.result
-      assert_equal(['foo', nil, true], r)
+    [
+      [
+        'array',
+        '["foo",null,true]',
+        '["foo",null,true]'
+      ],
+      [
+        'truncated_array',
+        '["foo",nul',
+        '["foo"]'
+      ],
+      [
+        'truncated_number_in_array',
+        '["foo",1',
+        '["foo"]'
+      ],
+      [
+        'truncated_array_ends_with_comma',
+        '["foo",',
+        '["foo"]'
+      ],
+      [
+        'truncated_array_ends_with_opening_square_bracket',
+        '["foo",[',
+        '["foo",[]]'
+      ],
+      [
+        'does_not_touch_uncorrupted_number_in_array',
+        '["foo",1]',
+        '["foo",1]'
+      ],
+      [
+        'truncated_value_in_nested_array',
+        '["foo",[10.3,["bar",fals',
+        '["foo",[10.3,["bar"]]]'
+      ],
+      [
+        'object',
+        '{"foo":{"bar":null}}',
+        '{"foo":{"bar":null}}'
+      ],
+      [
+        'truncated_object',
+        '{"foo":true,"bar":fals',
+        '{"foo":true}'
+      ],
+      [
+        'truncated_first_value_in_object',
+        '{"foo":tru',
+        '{}'
+      ],
+      [
+        'truncated_number_in_object',
+        '{"foo":13.5,"bar":1',
+        '{"foo":13.5}'
+      ],
+      [
+        'does_not_touch_uncorrupted_number_in_object',
```

Comment:
There's a mixture of `truncated`/`corrupted` in these test case names. It doesn't change anything about the code so this is the lowest priority, but they feel like they could be consistent